### PR TITLE
docs: AstraDB as Chat Memory

### PR DESCRIPTION
**Pull request by edwinjosechittilappilly**
Docs: update in Astra as chat memory.
This pull request includes an addition to the documentation to guide users on how to store chat memory in a database. The new section provides detailed steps and an example flow to help users implement this feature.

Documentation update:

* [`docs/docs/Guides/guides-chat-memory.md`](diffhunk://#diff-8b4ae85f309e9cc3f51f766a1b3f4db8e08c75f9939d99a435f5458877f57489R54-R72): Added a new section titled "Chat Memory Storing in a Database" with steps to store and retrieve chat history from an external database using the `Chat Memory` and `Store Message` components, and an example flow diagram.

### CODE REVIEW

File: docs/docs/Guides/guides-chat-memory.md [hunk 1]
```diff
@@ -51,3 +51,43 @@ You can also display all messages stored across every flow and session by going
...
+### Connect the chat memory component to an external database
+
+1. Load the [Memory Chatbot](/starter-projects-memory-chatbot) starter project.
+This starter project extends the basic prompting flow to include a chat memory component.
+2. Add the [Store Message](/components-helpers#store-message) component to the flow.
+The **Store message** component stores messages in the external database.
+3. Add the [AstraDBChatMemory Component](/Components/components-memories#astradbchatmemory-component) to the flow.
+The **Astra DB Chat Memory** component stores and retrieves messages from **Astra DB**.
+4. Configure the **AstraDBChatMemory** component with your AstraDB instance details.
+	1. In the **Astra DB Application Token** field, add your Astra token.
```
Comment (mendonk) [inline docs/docs/Guides/guides-chat-memory.md:82]:
```suggestion
	1. In the **Astra DB Application Token** field, add your Astra token. (`AstraCS:...`)
```

File: docs/docs/Guides/guides-chat-memory.md [hunk 2]
```diff
@@ -51,3 +51,43 @@ You can also display all messages stored across every flow and session by going
...
+	2. In the **API Endpoint** field, add your Astra database's endpoint.
```
Comment (mendonk) [inline docs/docs/Guides/guides-chat-memory.md:83]:
```suggestion
	2. In the **API Endpoint** field, add your Astra database's endpoint. (`https://12adb-bc-5378c845f05a6-e0a12-bd889b4-us-east-2.apps.astra.datastax.com`)
```

File: docs/docs/Guides/guides-chat-memory.md [hunk 3]
```diff
@@ -51,3 +51,43 @@ You can also display all messages stored across every flow and session by going
...
+
+Chat memory is stored to an external database or vector store using the [Store Message](/components-helpers#store-message) component.
+
+The [**Chat Memories**](/Components/components-memories) components provide access to their respective external databases **as memory**. This allows AIs to access external memory for persistence and context retention. For example, connect the **Chat Memory** component to an **AstraDBChatMemory*** component to store the message history in an external Astra DB database.
+
+This example stores and retrieves chat history from an [AstraDBChatMemory](/Components/components-memories#astradbchatmemory-component) component with **Store Message** and **Chat Memory** components.
+
+### Prerequisites
+
+* [OpenAI API key](https://platform.openai.com/)
```
Comment (KimberlyFields) [inline docs/docs/Guides/guides-chat-memory.md:68]:
```suggestion
* [An OpenAI API key](https://platform.openai.com/)
```

File: docs/docs/Guides/guides-chat-memory.md [hunk 4]
```diff
@@ -51,3 +51,43 @@ You can also display all messages stored across every flow and session by going
...
+* [An Astra DB vector database created](https://docs.datastax.com/en/astra-db-serverless/get-started/quickstart.html) with:
```
Comment (KimberlyFields) [inline docs/docs/Guides/guides-chat-memory.md:69]:
```suggestion
* [An Astra DB vector database](https://docs.datastax.com/en/astra-db-serverless/get-started/quickstart.html) with:
```

**Review by KimberlyFields — APPROVED** [review]
Made a few suggestions, but approving.

**Pull request merged**